st2common/st2common/util/virtualenvs.py

```diff
@@ -182,13 +184,27 @@ def remove_virtualenv(virtualenv_path, logger=None):
     return True
 
 
-def install_requirements(virtualenv_path, requirements_file_path):
+def install_requirements(virtualenv_path, requirements_file_path, proxy_config=None):
     """
     Install requirements from a file.
     """
     pip_path = os.path.join(virtualenv_path, 'bin/pip')
-    cmd = [pip_path, 'install', '-U', '-r', requirements_file_path]
+    cmd = [pip_path]
+
+    if proxy_config:
+        cert = proxy_config.get('ca_bundle_path', None)
+        https_proxy = proxy_config.get('https_proxy', None)
+        http_proxy = proxy_config.get('http_proxy', None)
+
+        if http_proxy:
+            cmd.extend(['--proxy', http_proxy])
```

Comment:
In addion to `--proxy` flag, pip honors `http_proxy`, `https_proxy` and `no_proxy` env variables.

I'd like us to follow the same approch with ENVs.